Only call get_job_by_id when Required

jaredwebber opened the pull request:
## Tickets
Closes #82 
## Summary Of Changes
- Added check in `showDetailModal` which looks at whether the job being opened is populated with only minimum fields or if complete object exists
- Only calls backend `get_job_by_id` when job is minimally populated
## Notes for Reviewers
- Open up the network devtools tab and verify that `get_job_by_id` is only called when clicking on a job card for the first time
  - newly created jobs are fully populated in the frontend, so clicking the card for a job created in the same session should not post to backend

jaredwebber commented on frontend/src/views/JobTrackingView.vue:
```diff
@@ -158,13 +158,19 @@ export default {
-          job_id: job.id,
-        }).then((completeJob) => {
-          this.selectedJob = completeJob.job_data
+        if (!job.deadlines) {
```
this could be any of the non-minimal fields e.g. `cover_letter`, `description`, etc.

drkitt reviewed (changes requested):
It's possible to have a job that has only the minimal fields, so `get_job_by_id` gets called every time you click on that kind of job. To fix this, I think it should use a dedicated flag that tells whether a job has been accessed since the last time the page was loaded (like the original issue description), instead of checking any non-minimal fields.

jaredwebber commented:
> It's possible to have a job that has only the minimal fields, so `get_job_by_id` gets called every time you click on that kind of job. To fix this, I think it should use a dedicated flag that tells whether a job has been accessed since the last time the page was loaded (like the original issue description), instead of checking any non-minimal fields.

I updated it to check for undefined specifically which is what I meant to do - since even if a job has no deadlines, job.deadlines will be `null` or `[]` after the `get_job_by_id` is called. I also noticed that that I only updated the frontend objects when `save` is clicked, so I've updated that now. I think using a non-minimal field is potentially cleaner since we would have to remove a flag field when we post the job to the backend - open to other opinions tho.

drkitt reviewed (approved):
Works perfectly now

jaredwebber commented on frontend/src/views/JobTrackingView.vue:
```diff
@@ -158,13 +159,24 @@ export default {
+
+            this.selectedJob = completeJob.job_data
+            this.detailModalVisible = true
+          })
```
Sorted on edit & save - easier than in-order insertion since eslint doesn't allow parameter modification
